src/Engine/Audio/VorbisFile.hpp

```diff
@@ -3,43 +3,59 @@
 #include "SoundFile.hpp"
 #include "../linking.hpp"
 
+struct stb_vorbis;
+
 namespace Audio {
     /// Ogg Vorbis implementation of SoundFile.
     /**
      * Loads .ogg-files.
      */
     class VorbisFile : public SoundFile {
         public:
-            /// Loads the specified ogg-file.
-            /**
-             * @param filename Filename (relative or absolute) to ogg-file.
-             */
-            ENGINE_API explicit VorbisFile(const char* filename);
-            
+            /// Constructor.
+            ENGINE_API VorbisFile();
+
             /// Destructor.
             ENGINE_API ~VorbisFile() final;
             
             /// Get raw audio data.
             /**
-             * @return Raw audio data.
+             * @param offset Number of samples from start of file.
+             * @param samples Number of samples to read.
+             * @param data Address to store raw audio data.
+             * @return Number of valid samples loaded from file.
              */
-            ENGINE_API float* GetData() const final;
+            ENGINE_API int GetData(uint32_t offset, uint32_t samples, float* data) const final;
             
-            /// Get data size.
+            /// Get sample count.
             /**
-             * @return The length of the raw audio data.
+             * @return The number of samples in the raw audio data.
              */
-            ENGINE_API uint32_t GetSize() const final;
+            ENGINE_API uint32_t GetSampleCount() const final;
             
             /// Get sample rate.
             /**
              * @return The sound file's sample rate (Hz).
              */
             ENGINE_API uint32_t GetSampleRate() const final;
+
+            /// Get channel count.
+            /**
+             * @return The number of channels in audio.
+             */
+            ENGINE_API uint32_t GetChannelCount() const final;
             
         private:
-            float* data = nullptr;
-            int dataSize = 0;
+
+            /// Loads the specified ogg-file.
```

Comment:
Private methods shouldn't have doxygen comments. Also doesn't need to be exported.

Reply:
Fixed